qiskit/quantum_info/operators/symplectic/clifford.py

```diff
@@ -147,56 +173,102 @@ def __str__(self):
 
     def __eq__(self, other):
         """Check if two Clifford tables are equal"""
-        return super().__eq__(other) and self._table == other._table
+        return super().__eq__(other) and self._paulis == other._paulis
 
     # ---------------------------------------------------------------------
     # Attributes
     # ---------------------------------------------------------------------
     def __getitem__(self, key):
         """Return a stabilizer Pauli row"""
-        return self._table.__getitem__(key)
+        return self._paulis.__getitem__(key)
 
     def __setitem__(self, key, value):
         """Set a stabilizer Pauli row"""
-        self._table.__setitem__(key, value)
+        self._paulis.__setitem__(key, value)
 
     @property
+    def paulis(self):
+        """Return PauliList"""
+        return self._paulis
+
+    @paulis.setter
+    def paulis(self, value):
+        """Set the Pauli list"""
+        if not isinstance(value, PauliList):
+            value = PauliList(value)
+        self._paulis._z[:, :] = value._paulis._z
+        self._paulis._x[:, :] = value._paulis._x
+        self._paulis._phase[:] = value._paulis._phase
```

Comment:
I'm generally pretty against getter `property` methods doing non-trivial object creation, and if it's only a convenience getter, it doesn't need a setter - we need people to transition to the correct way of doing things (using `x` and `z`, or whatever).  Having a getter property like this means that a new Numpy array is created every time `x.array` is accessed, which is surprising, and it's often been a case of significant performance problems when writing code that consumes these objects.

Even something like `y = x.array[0] + x.array[1]` creates two new Numpy arrays, but there's no clue at the usage site that it would create even one.  This type of thing is usually safer as an actual method (that you call with `as_stabilizer_array()` or whatever), because it's clear when it's used that it's probably creating something and you shouldn't be calling it loads of times in a simple expression.

Reply:
Sorry. At first, I thought this pattern would be used frequently, so I added an `array` method, but when I tried it, I found that I rarely used it, so I decided to remove it.